python_ta/contracts/__init__.py

```diff
@@ -194,6 +196,19 @@ def wrapper(wrapped, instance, args, kwargs):
     return wrapper(wrapped)
 
 
+def _instance_init_in_callstack(instance) -> bool:
+    """Return whether instance's init is part of the current callstack
+    """
+    frame = inspect.currentframe().f_back
+    while frame:
+        frame_context_name = inspect.getframeinfo(frame).function
+        frame_context_self = frame.f_locals.get('self', None)
```

Comment:
You can be a bit more specific here and check that the first parameter name `== 'self'` as well.

That said, there's still the possibility of having a method which is called from a top-level function whose first parameter is `self`, which would cause contract checking to be skipped. This is a bit subtle, but probably worth documenting. Please add a note about this to the documentation page for PyTA's contract checking.